web/src/server/api/services/EventProcessor.ts

```diff
@@ -191,6 +192,28 @@ export class ObservationProcessor implements EventProcessor {
           )
         : [undefined, undefined];
 
+    const newTotalCount =
+      "usage" in this.event.body
+        ? this.event.body.usage?.total ??
+          (newInputCount ?? 0) + (newOutputCount ?? 0)
+        : undefined;
+
+    // Token costs
+    const userProvidedTokenCosts =
+      "usage" in this.event.body ? this.event.body?.usage ?? {} : {};
+
+    const tokenCounts = {
+      input: newInputCount,
+      output: newOutputCount,
+      total: newTotalCount,
+    };
+
+    const calculatedCosts = this.calculateTokenCosts(
+      internalModel,
+      userProvidedTokenCosts,
+      tokenCounts,
+    );
```

Comment:
I think we have the following cases here:
- user provided the cost in first observation, user provided no cost in second observation [+other way around]
- user provided no cost at all, model match in first observation, no model in the update event [+other way around]
- user provided no cost in first event (but we match the model and calculate everything) and provides cost in the second one [+ other way around]
- if no tokens and no cost and no model match, we dont do anything